File: example/1d-linear-convection/weno3/julia/03g/src/time_integration/rk1.jl
# src/time_integration/rk1.jl

mutable struct RK1Integrator
    base::TimeIntegratorBase
end

function RK1Integrator(cfd::Any)
    RK1Integrator(TimeIntegratorBase(cfd))
end

function step(integrator::RK1Integrator, dt::Float64)
    compute_residual(integrator.base)
    for i in integrator.base.domain.ist:(integrator.base.domain.ied - 1)
        j = map_idx(integrator.base, i)
        integrator.base.solution.u[i] += dt * integrator.base.solution.res[j]
    end
    apply_boundary(integrator.base)
    update_old_field(integrator.base.solution)
end

File: example/1d-linear-convection/weno3/python-v1/02a/src/time_integration/rk1.py
# time_integration/rk1.py

from .base import TimeIntegrator
from core.registry import register_component

@register_component('integrator', 'rk1')
class RK1Integrator(TimeIntegrator):
    """1阶显式欧拉（RK1）"""
    def step(self, dt):
        self.compute_residual()
        for i in range(self.domain.ist, self.domain.ied):
            j = self.map_idx(i)
            self.solution.u[i] += dt * self.solution.res[j]
        self.apply_boundary()
        self.solution.update_old_field()